gobblin-yarn/src/main/java/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -763,7 +770,7 @@ private void sendEmailOnShutdown(Optional<ApplicationReport> applicationReport)
     }
 
     try {
-      EmailUtils.sendEmail(ConfigUtils.configToState(this.config), subject, messageBuilder.toString());
+      emailSender.get().sendEmail(subject, messageBuilder.toString());
```

Comment:
You need to check if emailSender.isPresent() before a get(), else it will throw an exception: http://docs.guava-libraries.googlecode.com/git-history/v10.0.1/javadoc/com/google/common/base/Optional.html#get()